Controller/ThemeController.php

```diff
@@ -73,7 +88,13 @@ public function switchAction(Request $request)
 
         $this->activeTheme->setName($theme);
 
-        $url = $request->headers->get('Referer', '/');
+        if ($defaultRoute) {
```

Comment:
it would be good to not duplicate the get() call on the headers property. so i will either remove the 2nd parameter and then check for null in $url and in that case either use / or compute the route. or compute the fallback route before calling the get() method

Reply:
Done :)